src/mrpro/data/_KTrajectory.py

```diff
@@ -41,11 +43,99 @@ class KTrajectory:
 
     @property
     def broadcasted_shape(self) -> tuple[int, ...]:
-        """The broadcasted shape of the trajectory."""
+        """The broadcasted shape of the trajectory.
+
+        Returns
+        -------
+            broadcasted shape of trajectory
+        """
         shape = np.broadcast_shapes(self.kx.shape, self.ky.shape, self.kz.shape)
         return tuple(shape)
 
-    def as_tensor(self, stack_dim=0):
+    def _type_of_traj(
+        self, tolerance: float = 1e-2, orientation: Literal['kz-ky-kx', 'k2-k1-k0'] = 'kz-ky-kx'
+    ) -> list[TrajType]:
+        """Type of trajectory.
+
+        This function checks if the entries of the trajectory along certain dimensions
+            - are of shape 1 -> TrajType.SINGLEVALUE
+            - lie on a Cartesian grid -> TrajType.ONGRID
```

Comment:
does it check cartesian grid?
isnt it just regular spacing long the dimension?

Reply:
the values do not have to be regularly spaced but have to lie on a grid, i.e. [0 1 2 3 5 10] would also be ONGRID

Reply:
My question is regarding the 'Cartesian' here.
kz ky kx are by Definition Cartiesian. 
but do we check for perpendicularity in k2 k1 k0?
Cartesian would imply yes, but I don't see where, maybe I am missing it

Reply:
The way I see it is, that the points of the trajectory are always defined in the kz-ky-kx space. If we check the type along k2 we are checking if all the points along k2 lie on the kz-ky-kx grid.